src/main/java/de/zalando/aruha/nakadi/controller/EventPublishingController.java

```diff
@@ -61,14 +68,23 @@ public ResponseEntity postEvent(@PathVariable final String eventTypeName, @Reque
         try {
             final EventType eventType = eventTypeRepository.findByName(eventTypeName);
 
-            final JSONObject eventAsJson = parseJson(event);
+            try {
+                final Timer successfullyPublishedTimer = metricRegistry.timer(metricNameFor(eventTypeName, SUCCESS_METRIC_NAME));
```

Comment:
It will create a timer once and the MetricRegistry will return the same timer the next time. And I am not sure, as well.

We are measuring per event type. That's why the event type has to be taken into account and we cannot have one timer in the class. We would need a timer per event type. One possibility is to cache the timers. I was considering this but came to the conclusion that this is premature optimisation. Let's first try this out and improve it if necessary.